### ISSUE TITLE
Document Viewer | Annotation menu redesign

### ISSUE THREAD

**Issue by gnakm** [department-of-veterans-affairs/caseflow#2161]
**Description**
A major feedback we've received from users is that on a smaller screen, the comments are hard to reach. We added accordions to each section. 

## AC
- [ ] Verify that Document Information, Categories, Issue Tags and Comments are now accordions and should be expand and collapsable. Note: We don't have it in our Style Guide yet but we have a ticket #1482 and it follows the [pattern from USWDS ](https://standards.usa.gov/components/accordions/)
- [ ] Each accordion section should now have a 1px border of [gray-light](https://dsva-appeals-certification-dev-1895622301.us-gov-west-1.elb.amazonaws.com/styleguide#colors)
- [ ] The height of "hide menu" bar and accordion bars should match the height of top toolbar. 
- [ ] New Hide Menu bar color should be `$color-gray-warm-light` [See style guide colors](https://dsva-appeals-certification-dev-1895622301.us-gov-west-1.elb.amazonaws.com/styleguide#colors)
- [ ] By default, the sections should be expanded
- [ ] No space between hide menu bar and document info accordion
- [ ] There should be 14px spacing between edge accordion and content inside (i can provide mock)
- [ ] Remove `h5` Related issue with horizontal line and Comments with horizontal line
- [ ] When scrolled to the comments section, the accordion and add a comment button should be fixed and a double scroll bar should scroll all the comments below it
- [ ] When a user hits`alt+enter`, the comment accordion should expand and show the appropriate comment box highlighted. 

## Mock up

![image](https://user-images.githubusercontent.com/13420618/28286841-88418e1a-6b07-11e7-90ac-65c2fcf3edfc.png)

![image](https://user-images.githubusercontent.com/13420618/28286904-c2d89a28-6b07-11e7-9372-670cdfa4d9f5.png)

## Related
#1853 #2359 #1998 #2375

**Comment by abbyraskin** [comment]
@gnakm we should start finalizing designs for this -- receipt date edit will come later, but if we want to include any issues info from VACOLS, I think we should show that here.

**Comment by gnakm** [comment]
@abbyraskin Yup! Updated the mock. There are still pressing questions related to #1998 we should talk about it. Afterwards, I'll update this AC with new DAS stuff. 


Note: This was the previous mock.

![image](https://cloud.githubusercontent.com/assets/13420618/26472875/1f099678-4176-11e7-852f-7094d5057554.png)

**Comment by mmlumba** [comment]
@gnakm For the 4th AC you mention using `$color-gray-light` but the ticket you reference uses `$color-gray-warm-light`. Which color am I using for the Hide Menu bar?

**Comment by NickHeiner** [comment]
![image](https://user-images.githubusercontent.com/829827/28424150-e3f11eee-6d3a-11e7-846c-cba6ef4876b2.png)

@gnakm when the accordions are all collapsed, we don't want this vertical spacing between them, right?

**Comment by NickHeiner** [comment]
@gnakm If I hit `alt+c` to start placing a comment from the keyboard when the comment UI is in a closed accordion, we enter a weird state. The user hits `alt+enter` to place a comment, but there's no feedback that anything happened:

![comments with accordion collapsed](https://user-images.githubusercontent.com/829827/28424295-3d282e26-6d3b-11e7-83e9-eeecc8a3ecfa.gif)

(I can explain over videochat if the gif does not make it clear enough.)

**Comment by gnakm** [comment]
@mmlumba @NickHeiner No please. Every space we can save will be helpful!

**Comment by gnakm** [comment]
@NickHeiner +1 for being a thorough human being. 

Yeah I didn't think about that. When a user hits`alt+enter`, the comment accordion should expand and show the appropriate comment box highlighted. Adding that to the AC.

**Comment by NickHeiner** [comment]
The spacing is not going to be pixel-perfect to what is in the mocks. @gnakm is it ok if that's a follow-up ticket?

**Comment by kierachell** [comment]
### Do we need that extra scrollbar?
- Windows 7
- IE and Chrome
![whatamiscrolling](https://user-images.githubusercontent.com/18075411/28465911-765bb0b0-6df8-11e7-95b8-6934103d3e0b.gif)

![screen shot 2017-07-21 at 9 35 56 am](https://user-images.githubusercontent.com/18075411/28465906-7128a99a-6df8-11e7-81e2-c5150f36d8a3.png)


### There's some weird interaction between issue tag - clicking - accordion

- Start entering issue tag text (if the text is long enough it starts glitching out too)
  - also you get a bonus scrollbar...
- Click X to cancel(???) the creation
- Click to expand another accordion
- See the text come back?
![iguessyoucancallthisresponsive](https://user-images.githubusercontent.com/18075411/28466023-eb0ba9a6-6df8-11e7-8685-beb979f5054c.gif)

**Comment by astewarttistatech** [comment]
Looks like the spacing is off. @NickHeiner 

Env: UAT
Browsers: FF, Chrome, and IE

Steps to Validate: 
1. Access a document in Reader
2. Access the left side menu
3. Validate the accordions, their spacing to the start/end of content, as well as their color. 

Screenshots: 
<img width="1440" alt="screen shot 2017-07-26 at 7 20 51 am" src="https://user-images.githubusercontent.com/23038505/28619047-ad977700-71d4-11e7-982a-ec45a67d0545.png">
![readermenu](https://user-images.githubusercontent.com/23038505/28619064-bc2d8110-71d4-11e7-9c20-9cd9488cca0a.gif)

**Comment by NickHeiner** [comment]
@astewarttistatech, can you be more specific about what's wrong with the spacing?

@gnakm, do you think it's worth spending dev time to change the spacing from how it looks now?

**Comment by astewarttistatech** [comment]
Sorry, I thought the top screenshot made it clear. The top and bottom spacing of the content from the inside of the accordions are 14px instead of 20px. @NickHeiner

**Comment by gnakm** [comment]
@NickHeiner @astewarttistatech The 14px spacing is NBD. Let's go with it. Changing AC now.

**Comment by NickHeiner** [comment]
Thanks @gnakm.

@astewarttistatech, it was unclear to me. I saw the `14px` measurement in your screenshot. But that measurement was measuring a different stretch than what Gina had measured in the AC. The `14px` is the yellow section in this screenshot:

![image](https://user-images.githubusercontent.com/829827/28677165-afc01ffa-72ba-11e7-9e5e-ac6f47345d54.png)

But the `14px` does not include the blue section. The AC was:

![image](https://user-images.githubusercontent.com/829827/28677187-ba63bab6-72ba-11e7-97e3-6a2abf57b637.png)

Your measurement did not extend as far as Gina's measurement. And when I did a measurement that tried to extend as far as Gina's, I did get `20px`.

![image](https://user-images.githubusercontent.com/829827/28677310-25004fd8-72bb-11e7-8e25-334acf012fec.png)

So, that's why I was unsure what the problem was. :smile:
